backend/modules/users.js

```diff
@@ -48,16 +48,48 @@ import pool from '../utils/database.js';
 
 const router = Router();
 
+//Login via username and password
 router.post('/login', async (req, res) => {
   try {
-    if (!req.user) {
-      return res.status(401).json({success: false, message: 'No token provided'});
+    const { username, password, token } = req.body;
+
+    if (token) {
+      // Validate the token
+      const user = await getUser(token);
+      if (user) {
+        return res.status(200).json({ success: true, message: 'Token is valid', user });
+      } else {
+        return res.status(401).json({ success: false, message: 'Invalid token' });
+      }
+    }
+
+    if (user.password !== password) {
+      return res.status(401).json({ success: false, message: 'Invalid username or password' });
+    }
+
+    if (!user.enabled) {
+      return res.status(403).json({ success: false, message: 'User is disabled' });
     }
 
-    res.status(200).json({success: true, message: 'Login successful', data: req.user});
+    // Generate a token
+    const token = uuidv4();
+
+   //token update
+    await pool.query('UPDATE "public"."users" SET "token" = $1 WHERE "id" = $2', [token, user.id]);
+
+    // Storing info in tokenMap
+    tokenMap[token] = {
+      id: user.id,
+      name: user.name,
+      pfp: user.pfp,
+      role: user.role,
+    };
+
+    res.status(200).json({ success: true, message: 'Login successful', token });
```

Comment:
And return back `res.status(200).json({ success: true, message: 'Login successful', data: token });`